### PR TITLE
Fixes #394 & #634

### LINKED ISSUE #394: layout.Tile border_focus gives NameError if modified
When setting `layout.Tile` with `border_focus` argument, i.e. with:
```python
layout.Tile(ratio=0.5, border_focus="ffffff") 
```
Qtile crashes and says:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/libqtile/manager.py", line 585, in _xpoll
    r = h(e)
  File "/usr/lib/python2.7/dist-packages/libqtile/manager.py", line 892, in handle_UnmapNotify
    self.unmanage(e.window)
  File "/usr/lib/python2.7/dist-packages/libqtile/manager.py", line 406, in unmanage
    c.group.remove(c)
  File "/usr/lib/python2.7/dist-packages/libqtile/group.py", line 215, in remove
    self.focus(nextfocus, True)
  File "/usr/lib/python2.7/dist-packages/libqtile/group.py", line 156, in focus
    l.focus(win)
  File "/usr/lib/python2.7/dist-packages/libqtile/layout/max.py", line 46, in focus
    self.group.layoutAll()
  File "/usr/lib/python2.7/dist-packages/libqtile/group.py", line 92, in layoutAll
    self.layout.layout(normal, screen)
  File "/usr/lib/python2.7/dist-packages/libqtile/layout/base.py", line 54, in layout
    self.configure(i, screen)
  File "/usr/lib/python2.7/dist-packages/libqtile/layout/tile.py", line 189, in configure
    bc = self.group.qtile.colorPixel(self.border_focus)
  File "/usr/lib/python2.7/dist-packages/libqtile/utils.py", line 99, in wrap
    ret = f(self, *args)
  File "/usr/lib/python2.7/dist-packages/libqtile/manager.py", line 364, in colorPixel
    return self.conn.screens[0].default_colormap.alloc_color(name).pixel
  File "/usr/lib/python2.7/dist-packages/libqtile/xcbq.py", line 288, in alloc_color
    self.cid, len(color), color
BadName: <xcb.xproto.NameError object at 0x319c908>
```

### PR DESCRIPTION
**Pull request by aniruddhkanojia**
This fixes #394 and #634 
Added changes as asked by @tych0 .
Also fixed the tests.

Please review.

### CODE REVIEW

File: libqtile/xcbq.py [hunk 1]
```diff
@@ -37,6 +37,7 @@
 from __future__ import print_function, division
 
 import six
+import re
```
Comment (kynikos) [inline libqtile/xcbq.py]: This import isn't necessary anymore I guess

File: libqtile/xcbq.py [hunk 2]
```diff
@@ -316,20 +317,20 @@ def alloc_color(self, color):
...
         """
-        if color.startswith("#"):
-            if len(color) != 7:
+        try:
+            return self.conn.conn.core.AllocNamedColor(
+                self.cid, len(color), color
+            ).reply()
+        except xcffib.xproto.NameError:
+            if len(color) < 6:
                 raise ValueError("Invalid color: %s" % color)
```
Comment (kynikos) [inline libqtile/xcbq.py]: I wonder if we really need this test... there can be more things wrong in the `color` string than its length, and IMHO the built-in exception that would be normally raised in those cases would be enough.

File: libqtile/xcbq.py (hunk 1 above)
Comment (aniruddhkanojia) [inline libqtile/xcbq.py]: ok. should I make another PR ?

File: libqtile/xcbq.py (hunk 2 above)
Comment (kynikos) [inline libqtile/xcbq.py]: Sorry, I'm referring to the `if len(color) < 6:` test.

Comment (aniruddhkanojia) [inline libqtile/xcbq.py]: I used it to avoid list index out of error.

Comment (aniruddhkanojia) [inline libqtile/xcbq.py]: What do you recommend ?

File: libqtile/xcbq.py (hunk 1 above)
Comment (kynikos) [inline libqtile/xcbq.py]: As I said, I'm not an official maintainer, however as you can see the new commit is already visible in this PR, so there's no need to open a new one. tych0 will squash the commits as needed if he thinks it's appropriate.

File: libqtile/xcbq.py (hunk 2 above)
Comment (kynikos) [inline libqtile/xcbq.py]: The fact is that you're avoiding `IndexError` just to raise `ValueError` :) If it were me, I'd just remove the test and the `ValueError` line, and let Python raise what's to be raised as needed, but you may want to wait for tych0 for an official opinion.

Comment (aniruddhkanojia) [inline libqtile/xcbq.py]: I see your point.
I think I will wait for tych0 and hope he is ok with it .
I have spent a lot of unnecessary time on this little Bug :stuck_out_tongue:

Comment (kynikos) [inline libqtile/xcbq.py]: Yep, maybe `except xcffib.xproto.NameError:` could just `pass`, and its block could be enclosed in another `try` that catches IndexError and other possible exceptions caused by bad `color` strings, for example containing non-hexadecimal characters (which would break the `int` calls I guess), but this is all fussy optimization after all, I agree ;)

Comment (aniruddhkanojia) [inline libqtile/xcbq.py]: Fingers Crossed :sweat_smile:

Comment (tych0) [inline libqtile/xcbq.py]: I do think the changes @kynikos has suggested would make this a little more readable, if you can make those the rest of it looks good to me.

Comment (aniruddhkanojia) [inline libqtile/xcbq.py]: @tych0  is it ok  now ?

**Comment by aniruddhkanojia** [comment]
Closing this as is is duplicate of #638 and #640.
